crates/wallet/src/wallet/params.rs

```diff
@@ -32,22 +32,38 @@ where
 pub struct CreateParams {
     pub(crate) descriptor: DescriptorToExtract,
     pub(crate) descriptor_keymap: KeyMap,
-    pub(crate) change_descriptor: DescriptorToExtract,
+    pub(crate) change_descriptor: Option<DescriptorToExtract>,
     pub(crate) change_descriptor_keymap: KeyMap,
     pub(crate) network: Network,
     pub(crate) genesis_hash: Option<BlockHash>,
     pub(crate) lookahead: u32,
 }
 
 impl CreateParams {
-    /// Construct parameters with provided `descriptor`, `change_descriptor` and `network`.
+    /// Construct parameters with provided `descriptor`.
+    pub fn new_single_descriptor<D: IntoWalletDescriptor + 'static>(descriptor: D) -> Self {
+        Self {
+            descriptor: make_descriptor_to_extract(descriptor),
+            descriptor_keymap: KeyMap::default(),
+            change_descriptor: None,
+            change_descriptor_keymap: KeyMap::default(),
+            network: Network::Bitcoin,
+            genesis_hash: None,
+            lookahead: DEFAULT_LOOKAHEAD,
+        }
+    }
```

Comment:
Instead of having a new constructor, I would prefer changing `new` to only take in a single `descriptor` and creating:
```rust
pub fn change_descriptor(mut self, change_descriptor: D) -> Self { todo!() }
```

Ergonomically, maybe we should have two methods for `keymap`, one for each keychain?

Reply:
That sounds reasonable.

cc @notmandatory

Reply:
If we change `CreateParams::new` to take a single descriptor (right now takes 2) it will  be a breaking change.
I think that it is acceptable and the time to do it is now!

Reply:
If you are considering a breaking change, then IMO the cleanest way to do it is similar to the [Wallet::load()](https://docs.rs/bdk_wallet/1.0.0-beta.1/bdk_wallet/struct.Wallet.html#method.load) API, with a single method `descriptors()` that takes an optional change_descriptor.

```rust
pub fn descriptors(mut self, descriptor: D, change_descriptor: Option<D>) -> Self { todo!() }
```

Reply:
@thunderbiscuit but that means we need an error case when `descriptor` is not provided.

Reply:
Sounds good thanks for expanding on the constructor design @evanlinjin I had not seen that. Cheers! I have more questions which I'll ask in the thread below.

Reply:
> Instead of having a new constructor, I would prefer changing `new` to only take in a single `descriptor` and creating ...

Since our recommended practice is to provide two descriptors I'd rather leave `new(...)` as is and only add `new_single_descriptor(...)` as @ValuedMammal has it for the few projects that need it.

> Ergonomically, maybe we should have two methods for `keymap`, one for each keychain?

I'd rather not have separate methods since it's an API change and doesn't add very much.

Reply:
> I think it makes sense for the builder struct to take in all that is necessary in `new`, and all that is optional with additional methods. Hence, my original suggestion.

I generally agree with this and do think it would be cleaner than having two `new` constructors but I don't have a strong preference in terms of API

Reply:
After more discussion I think it's better to be explicit and provide another constructor along the lines of `new_single_descriptor`. This also makes for a convenient place to add docs that are specific to single keychain wallets.